### PR TITLE
Add BOM for dependency management

### PR DESCRIPTION
**Pull request by lal-s**
What was changed? Why is this necessary?
----------------------------------------

Add Bill of Materials for dependency management.

How was it tested?
----

- Tested the BOM against a service

- Tested the archetype making sure it generates a service which uses BOM in dependency management
 

How to test
----

*This is bare minimum acceptable testing*

- [ ] `mvn clean install -U`

Reviewers
----

- [x] [John Leacox](https://github.com/johnlcox)
- [x] [Sundeep Paruvu](https://github.com/sparuvu)
- [x] [Nimesh Subramanian](https://github.com/nimeshsubramanian)
- [x] [Brian van de Boogaard](https://github.com/b-boogaard)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,11 +2,14 @@
 
 ## 2.5 - ???
 
+### Enhancements
```
Comment (johnlcox) [inline CHANGELOG.md]: I would consider this an addition which is a section we have sometimes when something totally new is added.

File: bom/pom.xml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>beadledom-parent</artifactId>
+        <groupId>com.cerner.beadledom</groupId>
+        <version>2.5-SNAPSHOT</version>
+    </parent>
+
+    <groupId>com.cerner.beadledom</groupId>
```
Comment (johnlcox) [inline bom/pom.xml]: `groupId` should be unnecessary since it's the same as the parent `groupId`.

File: CHANGELOG.md (hunk 1 above)
Comment (lal-s) [inline CHANGELOG.md]: [eae7681](https://github.com/cerner/beadledom/commit/eae76819dd8eb9035160420564c88d8d716148c7)

File: bom/pom.xml (hunk 2 above)
Comment (lal-s) [inline bom/pom.xml]: [eae7681](https://github.com/cerner/beadledom/commit/eae76819dd8eb9035160420564c88d8d716148c7)

**Comment by johnlcox** [comment]
+1

File: pom.xml [hunk 3]
```diff
@@ -33,13 +33,14 @@
             <url>https://github.com/b-boogaard</url>
         </developer>
         <developer>
-            <name>Supryia Lal</name>
+            <name>Supriya Lal</name>
```
Comment (b-boogaard) [inline pom.xml]: Whoops

**Comment by b-boogaard** [comment]
+1

File: pom.xml [hunk 4]
```diff
@@ -63,6 +64,8 @@
         <module>configuration</module>
 
         <module>client</module>
+
+        <module>bom</module>
```
Comment (johnlcox) [inline pom.xml]: I was thinking that too. Should we change it here or as another PR?

Comment (sparuvu) [inline pom.xml]: I think it's okay to do it as part of this issue, as this is pretty trivial.

Comment (lal-s) [inline pom.xml]: [adefe1a](https://github.com/cerner/beadledom/commit/adefe1a0359dd834a62e8c2054eacf6c3e6245e6)

**Pull request merged**